user/nix/template.py

```diff
@@ -0,0 +1,126 @@
+    # and history don't work with readline. As libedit has a readline compatible
+    # interface, history and completion won't work with libedit either. Patches
+    # to be able to use libedit instead of readline are in `patches/`, but there
+    # will be missing functionality, like with readline.
+    "-Dlibcmd:readline-flavor=editline",
+    # We need the below to fix "ERROR: clang does not know how to do prelinking.",
+    # a.k.a. nix only supports shared libraries.
+    "-Ddefault_library=shared",
+    # We want man pages.
+    "-Ddoc-gen=true",
+    # We don't need the perl bindings (and don't have the dependencies to build it).
+    "-Dbindings=false",
+    # We don't do test as it requires network, so don't build them.
+    "-Dunit-tests=false",
+]
+hostmakedepends = [
+    "meson",
+    "bison",
+    "flex",
+    "cmake",
+    "pkgconf",
+    "bash",
+    # These are needed for the doc-gen target.
+    "doxygen",
+    "mdbook",
+    # Preprocessor for mdbook that checks for broken links
+    # The docs include a lot of things, but we only package the man pages.
+    # man pages don't support links, so we don't care if they are dead or not.
+    # "mdbook-linkcheck",
+    "rsync",
+    "jq",
+]
+makedepends = [
+    "boost-devel",
+    "openssl3-devel",
+    "libarchive-devel",
+    "libsodium-devel",
+    "brotli-devel",
+    "nlohmann-json",
+    "curl-devel",
+    "libseccomp-devel",
+    "sqlite-devel",
+    "gc-devel",
+    "lowdown-devel",
+    "libgit2-devel",
+    "toml11",
+    # Use editline as nix recommends, but readline or libedit also work.
+    "editline-devel",
+]
+# We don't actually use these at the moment, because checks are disabled.
+checkdepends = [
+    "rapidcheck",  # not in cports yet
+    "gtest-devel",
+]
+pkgdesc = "Purely functional package manager"
+maintainer = "Mathijs Rietbergen <mathijs.rietbergen@proton.me>"
+license = "LGPL-2.1-or-later"
+url = "https://github.com/NixOS/nix"
+source = f"{url}/archive/refs/tags/{pkgver}.tar.gz"
+sha256 = "283ac42ec602e335ffd9c42721f35f1a20e8a7eb629ec5d9cc604a2fc2489ec0"
+# With `int` enabled, all nix tools constantly fail with "Illegal instruction".
+hardening = ["!int"]
+# Checks require network to download from nixpkgs, but network is not available
+# in check phase.
+# Cross compiling fails as meson cannot find CMake when cross compiling.
+# This seems like a generic issue, unrelated to nix...
+options = ["!check", "!cross"]
+
+
+def post_install(self):
+    # Remove systemd files.
+    self.uninstall("usr/lib/systemd")
+
+    # Creating /nix/var/nix/daemon-socket/socket is unneeded as nix-daemon can
+    # do it on its own, but the systemd unit files require the socket to be
+    # present before starting (why? I have no clue).
+    # That is why it exists upstream. We don't need this by any means.
+    self.uninstall("usr/lib/tmpfiles.d")
+
+    # We have our own, so we don't use these.
```

Comment:
you put all this verbosity in the comments but then don't state important stuff like *why* custom profile is shipped